File: YourChoice.Web/ClientApp/src/app/modelo/produto.ts
import { Ingrediente } from './ingrediente';
import { ProdutoIngrediente } from './produtoIngrediente';
export class Produto {
  id: number;
  nome: string;
  descricao: string;
  preco: number;
  nomeArquivo: string;
  precoOriginal: number;
  quantidade: number;
  ingredientesProdutos: Ingrediente[];
  produtoIngredientes: ProdutoIngrediente[];

  constructor() {
    /*this.produtoIngredientes = [];*/
    this.ingredientesProdutos = [];
  }

}

File: YourChoice.Dominio/Entidades/Produto.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace YourChoice.Dominio.Entidades
{
    public class Produto : Entidade
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string Descricao { get; set; }
        public decimal Preco { get; set; }
        public string NomeArquivo { get; set; }
        public virtual ICollection<ProdutoIngrediente> ProdutoIngrediente { get; set; }
        [NotMapped]
        public virtual ICollection<Ingrediente> IngredientesProdutos { get; set; }
        public override void Validate()
        {
            if (string.IsNullOrEmpty(Nome))
                AdicionarCritica("Nome do produto não foi informado");

            if (string.IsNullOrEmpty(Descricao))
                AdicionarCritica("Descrição não foi informado");
        }
    }
}
